### ISSUE TITLE
Line break in request URL causing request without params instead of throwing an error

### ISSUE THREAD

**Issue by psylence303** [a85/POSTMan-Chrome-Extension#4267]
## App Details:

```
Postman for Mac
Version 6.0.7
OS X 17.4.0 / x64
```

## Issue Report:
 1. Did you encounter this recently, or has this bug always been there:

Recently.

 2. Steps to reproduce/expected behavior:

I will use https://jsonplaceholder.typicode.com/ to illustrate the behavior. The request URL is `https://jsonplaceholder.typicode.com/comments?postId=2`.

If we will add a line break so the URL will look like:

`https://jsonplaceholder.typicode.com`
`/comments?postId=2`

It will fail with the expected error:

_There was an error connecting to https://jsonplaceholder.typicode.com%0A/comments?postId=2._

However, if we will add a line break after the trailing slash of the base URL:

`https://jsonplaceholder.typicode.com/`
`comments?postId=2`

Postman will not throw an error and will proceed with the request, however, it will ignore the `postId=2` parameter (and, from my experience, generally any request params):

`GET /comments HTTP/1.1
Host: jsonplaceholder.typicode.com
Cache-Control: no-cache
Postman-Token: fd83f8d6-07ea-4ed8-a4f7-2de3a071a855`

Expected behavior, in this case, is either:
 1) throwing the error because URL is actually broken
 2) proceeding with the request but with the params applied

Ok, I know that it is highly unlikely that somebody will accidentally add a line break to the URL bar of Postman (at least because it is possible only by copy/pasting the line break). However, it is quite easy to add the line break accidentally in the env variable section and it is not that visible there. That's the way I found this issue, I had a line break in the env variable with the base URL of my API and was wondering why the request returns different data from what was expected.

 3. Console logs (http://blog.getpostman.com/2014/01/27/enabling-chrome-developer-tools-inside-postman/ for the Chrome App, View->Toggle Dev Tools for the Mac app):

Logs are not showing anything special besides usual request/response logging:

`SyncChangesetEventService subscription Object {data: Object, name: "sent-to-server", namespace: "sync-changeset"}`

**Comment by prashantagarwal** [comment]
@psylence303 Thanks for reporting this. Just curious, how were you able to add a line break. Pressing Enter in the URL should send the request.

**Comment by psylence303** [comment]
@prashantagarwal it was not in the URL bar but was a part of the environment variable (it is fairly easy to add it there by pressing 'Enter' or copy/pasting from somewhere else).

**Comment by shamasis** [comment]
Some notes:

new-line is not allowed in domain so, we need to choose an appropriate behaviour as to whether this needs to be encoded or requesting be blocked

1. In "path" part of the url, this can be tacked with encoding and is already handled (something else is breaking here.)
2. In domain part, no encoding can handle this (so we have to error out)

Now, having said that... I ran a few tests:

1. newline before slash and after domain
> got response
> call went out as `GET /%0A/comments?postId=2 HTTP/1.1`

2. newline within path
> got 404
> since parts of path was stripped `GET /ments HTTP/1.1` 

3. newline in query param
> got response
> `GET /comments?post%0AId=2 HTTP/1.1`

In conclusion we need to do the following:
1. Look into path parsing regular expression in url parser to split accurately even with newlines (current encoding takes care of their presence anyway) @codenirvana 
2. Fix the inconsistency in UI that allows newline in the environment editor but not in preview @vkaegis

**Comment by shamasis** [comment]
@psylence303 ... in latest Postman Canary, this is handled differently and even UI hints are added to make new lines and white spaces more visible. More details at https://github.com/postmanlabs/postman-app-support/issues/8154

**Comment by codenirvana** [comment]
This is fixed in the latest version of the Postman app (v7.20.0) by enabling the "Use next generation URL processing" setting as mentioned in #8154.

Feel free to leave your feedback at #8154 or reopen this thread if the issue persists.
